Modules/IO/NIFTI/src/itkNiftiImageIO.cxx

```diff
@@ -1759,14 +1758,15 @@ IsAffine(const mat44 & nifti_mat)
       return false;
     }
   }
-  // Get the inverse of this matrix:
-  // Make sure the matrix is invertible to begin with...
-  if (fabs(vnl_determinant(mat)) <= 1e-5)
+
+  const float condition = vnl_matrix_inverse<float>(mat.as_matrix()).well_condition();
+  // Check matrix is invertible by testing condition number of inverse
+  if (!(condition > std::numeric_limits<float>::epsilon()))
```

Comment:
Why ! > instead of <= ?

Reply:
`! >` and `<=` give different results for floating point NaNs.  With the former, if the condition is small or is a NaN then ...

Reply:
Right! Pretty subtle though. Something like `if (!isfinite(condition) || (condition < epsilon))` might make the intention clearer.

Reply:
@seanm you're right, that is a nicer way to phrase it.

Reply:
The `! >` approach dates back forever and its purpose is easily recognizable by us oldtimers.  If we are nonetheless worried that it is opaque, that could be addressed in a comment.

If we instead go with the newer approach, instead of `isfinite` we might want to be checking `isnan` -- unless we are sure that a positive infinity should return `false` in our current context, or unless we are sure that positive infinity is impossible in our current context.

Reply:
>The ! > approach dates back forever and its purpose is easily recognizable by us oldtimers

Maybe I'm an even older timer than you then, because my foggy old memory forgot about this trick. :)